gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/DateTime.java

```diff
@@ -34,7 +32,7 @@
  * @see <a href="https://cloud.google.com/datastore/docs/concepts/entities">Google Cloud Datastore
```

Comment:
Datastore says that precision is still limited to microseconds.  I will check to see whether there's any chance for nanos to be set to some garbage, which could cause microseconds to be off by one.